src/scanner.cc

```diff
@@ -701,31 +732,43 @@ class Scanner
         {
             advance(lexer);
 
+            // empty attached modifier
             if (lexer->lookahead == found_attached_modifier->first)
             {
                 advance(lexer);
                 return NONE;
             }
 
+            /* TODO: what is this?
             auto can_have_modifier = [&]()
             {
                 return !m_ActiveModifiers[(VERBATIM_OPEN - BOLD_OPEN) / 2];
             };
+            */
 
-            if (!std::iswspace(lexer->lookahead) && !m_ActiveModifiers[(found_attached_modifier->second - BOLD_OPEN) / 2])
+            if (
+                (!std::iswspace(lexer->lookahead) || (m_ActiveModifiers[(RANGED_MODIFIER_OPEN - BOLD_OPEN) / 2]))
+                && !m_ActiveModifiers[(found_attached_modifier->second - BOLD_OPEN) / 2]
+            )
             {
-                if (can_have_modifier())
-                    m_ActiveModifiers.set((found_attached_modifier->second - BOLD_OPEN) / 2);
+                // if (can_have_modifier())
```

Comment:
It was used here but it was causing issues when implementing ranged modifiers and removing it didn't seem to break anything..